### PR TITLE
【PaddlePaddle Hackathon 2】29、为 Paddle 新增 PixelUnshuffle 组网 API

### PR DESCRIPTION
**Pull request by BrilliantYuKaimin**
增加nn.PixelUnshuffle和nn.functional.pixel_unshuffle的中文文档。

该算子的实现PR为https://github.com/PaddlePaddle/Paddle/pull/40728

PADDLEPADDLE_PR=40728

### CODE REVIEW

File: docs/api/paddle/nn/Overview_cn.rst [hunk 1]
```diff
@@ -267,6 +267,7 @@ Vision层
     
 
     " :ref:`paddle.nn.PixelShuffle <cn_api_nn_PixelShuffle>` ", "将一个形为[N, C, H, W]或是[N, H, W, C]的Tensor重新排列成形为 [N, C/r**2, H*r, W*r]或 [N, H*r, W*r, C/r**2] 的Tensor"
+    " :ref:`paddle.nn.PixelUnshuffle <cn_api_nn_PixelUnshuffle>` ", "PixelShuffle的逆操作，将一个形为[N, C, H, W]或是[N, H, W, C]的Tensor重新排列成形为 [N, C*r*r, H/r, W/r] 或 [N, H/r, W/r, C*r*r] 的Tensor"
```
Comment (TCChenlong) [inline docs/api/paddle/nn/Overview_cn.rst:270]: Please add the index of `pixel_unshuffle` below

File: docs/api/paddle/nn/PixelUnshuffle_cn.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - **downscale_factor** (int): 减小空间分辨率的减小因子。
+    - **data_format** (str，可选): 数据格式，可选："NCHW"或"NHWC"，默认:"NCHW"。
+    - **name** (str，可选): 操作的名称(可选，默认值为None)。更多信息请参见 :ref:`api_guide_Name`。
+
+形状
+:::::::::
+    - **x** (Tensor): 默认形状为（批大小，通道数，高度，宽度），即NCHW格式的4-D Tensor或NHWC格式的4-DTensor。 其数据类型为float32, float64。
+    - **output** (Tensor): 默认形状为（批大小，通道数，输出特征高度，输出特征宽度），即NCHW格式或NHWC的4-D Tensor。 其数据类型与输入相同。
+
+返回
```
Comment (TCChenlong) [inline docs/api/paddle/nn/PixelUnshuffle_cn.rst]: Can be omitted.

File: docs/api/paddle/nn/Overview_cn.rst (hunk 1 above)
Comment (TCChenlong) [reply docs/api/paddle/nn/Overview_cn.rst:270]: Please add here：https://github.com/PaddlePaddle/docs/blob/develop/docs/api/paddle/nn/Overview_cn.rst#%E5%85%AC%E7%94%A8%E6%96%B9%E6%B3%95

Comment (BrilliantYuKaimin) [reply docs/api/paddle/nn/Overview_cn.rst:270]: Done.

File: docs/api/paddle/nn/PixelUnshuffle_cn.rst (hunk 2 above)
Comment (BrilliantYuKaimin) [reply docs/api/paddle/nn/PixelUnshuffle_cn.rst]: Do you mean delete line 21 to 25?

Comment (dingjiaweiww) [reply docs/api/paddle/nn/PixelUnshuffle_cn.rst]: 26-28

Comment (BrilliantYuKaimin) [reply docs/api/paddle/nn/PixelUnshuffle_cn.rst]: done

**Review by dingjiaweiww — APPROVED** [review]

**Pull request merged**